drivers/sensor/hx711/hx711.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2020 Kalyan Sriram <kalyan@coderkalyan.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT avia_hx711
+
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#include "hx711.h"
+
+LOG_MODULE_REGISTER(HX711, CONFIG_SENSOR_LOG_LEVEL);
+
+static void hx711_convert_data(struct sensor_value *val, double raw_val,
+		double offset, double scale)
+{
+	double conv_val = (raw_val * scale) - offset;
+
+	val->val1 = (int32_t) conv_val;
+	val->val2 = (conv_val - ((int32_t) conv_val)) * 1e6;
+}
+
+static void hx711_convert_data_raw(struct sensor_value *val, double raw_val)
+{
+	val->val1 = (int32_t) raw_val;
+	val->val2 = (raw_val - ((int32_t) raw_val)) * 1e6;
+}
+
+static int hx711_channel_get(const struct device *dev,
+		enum sensor_channel chan,
+		struct sensor_value *val)
+{
+	struct hx711_data *drv_data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_FORCE:
+		hx711_convert_data(val, drv_data->data,
+				drv_data->offset, drv_data->scale);
+		break;
+	case SENSOR_CHAN_FORCE_RAW:
+		hx711_convert_data_raw(val, drv_data->data);
+		break;
+	default:
+		hx711_convert_data(val, drv_data->data,
+				drv_data->offset, drv_data->scale);
+		break;
+	}
+
+	return 0;
+}
+
+static int hx711_attr_set(const struct device *dev,
+		enum sensor_channel chan,
+		enum sensor_attribute attr,
+		const struct sensor_value *val)
+{
+	struct hx711_data *drv_data = dev->data;
+
+	switch (attr) {
+	case SENSOR_ATTR_OFFSET:
+		drv_data->offset = sensor_value_to_double(val);
+		break;
+	case SENSOR_ATTR_MULTIPLIER:
+		drv_data->scale = sensor_value_to_double(val);
+		break;
+	default:
+		drv_data->offset = sensor_value_to_double(val);
+		break;
+	}
+
+	return 0;
+}
+
+static int hx711_sample_fetch(const struct device *dev,
+		enum sensor_channel chan)
+{
+	struct hx711_data *drv_data = dev->data;
+	const struct hx711_config *cfg = dev->config;
+	uint8_t raw[4];
+	int32_t data = 0;
+
+	/* wait until the sensor is ready */
+	bool ready = false;
+
+	do {
+		int ret = gpio_pin_get(drv_data->dout, cfg->dout_pin);
+
+		if (ret < 0) {
+			LOG_ERR("Failed to read data sample.");
+			return -EIO;
+		}
+		ready = (ret == 0);
+	} while (!ready);
```

Comment:
Any timeout option here would be very useful in case of HW failure. Look [here](https://github.com/torvalds/linux/blob/eda809aef53426d044b519405d25d9da55319b76/drivers/iio/adc/hx711.c#L182) for reference:

Reply:
When tried it on nrf52840dk_nrf52840 with pins 1.02/1.03 as sck/dout, it spins here for ever. Moving to pins 0.26/0.27 solves the problem and it starts to work. Frankly not yet quite sure why these two legitimate pins (1.02/1.03) made an issue here, but its an example that wrong wiring or not attached HX711 chip isn't the only HW problem that will cause dead loop here.